internal/api/chisel.go

```diff
@@ -1,44 +1,80 @@
 package api
 
 import (
+	"context"
 	"fmt"
 	chclient "github.com/jpillora/chisel/client"
-	"github.com/kintohub/kinto-cli/internal/config"
+	"github.com/kintohub/kinto-cli/internal/types"
 	"github.com/kintohub/kinto-cli/internal/utils"
+	"io"
 	"strconv"
-	"sync"
 	"time"
 )
 
-func (a *Api) StartTeleport(blocksToForward []RemoteConfig) {
+func (a *Api) StartTeleport(blocksToForward []RemoteConfig, envId string, clusterId string) {
+
+	var host *types.TeleportResponse
+
+	// Default time to cancel is 30 minutes for our nginx gateway
+	// TODO: Move to env var / build arg
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Minute*30))
+	resp, err := a.getKubeCoreService(clusterId, envId).StartTeleport(
+		ctx, &types.TeleportRequest{EnvId: envId})
+	defer cancel()
+
+	if err != nil {
+		utils.TerminateWithError(err)
+	}
+
+	host, err = resp.Recv()
+
+	if err == io.EOF {
+		utils.TerminateWithCustomError("stream has no data!")
+	}
+	if err != nil {
+		utils.TerminateWithError(err)
+	}
 
 	var remotes []string
-	var err error
-	var wg sync.WaitGroup
 
 	for _, remote := range blocksToForward {
 		remotes = append(remotes, fmt.Sprintf(remote.FromHost+":"+strconv.Itoa(remote.FromPort)+
 			":"+remote.ToHost+":"+strconv.Itoa(remote.ToPort)))
 	}
 
 	chiselClient, err := chclient.NewClient(&chclient.Config{
-		KeepAlive:        time.Second,
-		MaxRetryInterval: time.Second,
-		Server:           config.ChiselHost,
+		MaxRetryInterval: 1 * time.Second,
+		MaxRetryCount:    50,
+		Server:           "https://" + host.Data.Host,
```

Comment:
action - use fmt.Sprintf("https://%s, host.Data.Host) instead. It's a go standard.